projects/igniteui-angular/src/lib/directives/text-highlight/text-highlight.directive.ts

```diff
@@ -333,9 +357,18 @@ export class IgxTextHighlightDirective implements AfterViewInit, AfterViewChecke
     public activateIfNecessary(): void {
         const group = IgxTextHighlightDirective.highlightGroupsMap.get(this.groupName);
         const column = group.columnIndex === undefined ? group.column : group.columnIndex;
-        const row = group.rowIndex === undefined ? group.row : group.rowIndex;
+        const row = group.rowIndex ? group.rowIndex : group.row;
```

Comment:
This change should be reverted since rowIndex=0 is a valid value for a row index but is is a falsy value. Anyway the rowIndex should be deprecated in another PR.